src/dispatch/function.py

```diff
@@ -23,12 +23,31 @@
 """
 
 
+# https://stackoverflow.com/questions/653368/how-to-create-a-decorator-that-can-be-used-either-with-or-without-parameters
+def decorator(f):
+    """This decorator is intended to declare decorators that can be used with
+    or without parameters. If the decorated function is called with a single
+    callable argument, it is assumed to be a function and the decorator is
+    applied to it. Otherwise, the decorator is called with the arguments
+    provided and the result is returned.
+    """
+
+    @wraps(f)
+    def method(self, *args, **kwargs):
+        if len(args) == 1 and len(kwargs) == 0 and callable(args[0]):
+            return f(self, args[0])
+        else:
+            return lambda realf: f(self, realf, *args, **kwargs)
```

Comment:
Lambda functions aren't serializable (can't be pickled). This may create issues if this code path is taken in a durable coroutine. An inline function def instead of a lambda will also probably cause issues because I don't think cell vars can be pickled either. We may need to add support for those in the experimental durable package.

Reply:
Ah! That explains some errors I saw at some point. Thanks for pointing it out 👍